polkadot-test-service/src/lib.rs

```diff
@@ -0,0 +1,316 @@
+		},
+		database: DatabaseConfig::RocksDb {
+			path: root.join("db"),
+			cache_size: 128,
+		},
+		state_cache_size: 16777216,
+		state_cache_child_ratio: None,
+		pruning: Default::default(),
+		chain_spec: Box::new(spec),
+		wasm_method: WasmExecutionMethod::Interpreted,
+		// NOTE: we enforce the use of the native runtime to make the errors more debuggable
+		execution_strategies: ExecutionStrategies {
+			syncing: sc_client_api::ExecutionStrategy::NativeWhenPossible,
+			importing: sc_client_api::ExecutionStrategy::NativeWhenPossible,
+			block_construction: sc_client_api::ExecutionStrategy::NativeWhenPossible,
+			offchain_worker: sc_client_api::ExecutionStrategy::NativeWhenPossible,
+			other: sc_client_api::ExecutionStrategy::NativeWhenPossible,
+		},
+		rpc_http: None,
+		rpc_ws: None,
+		rpc_ipc: None,
+		rpc_ws_max_connections: None,
+		rpc_cors: None,
+		rpc_methods: Default::default(),
+		prometheus_config: None,
+		telemetry_endpoints: None,
+		telemetry_external_transport: None,
+		default_heap_pages: None,
+		offchain_worker: Default::default(),
+		force_authoring: false,
+		disable_grandpa: false,
+		dev_key_seed: Some(key_seed),
+		tracing_targets: None,
+		tracing_receiver: Default::default(),
+		max_runtime_instances: 8,
+		announce_block: true,
+		base_path: Some(base_path),
+		informant_output_format,
+	})
+}
+
+/// Run a Polkadot test node using the Polkadot test runtime. The node will be using an in-memory socket, therefore you
+/// need to provide boot nodes if you want it to be connected to other nodes. The `storage_update_func` can be used to
+/// make adjustements to the runtime before the node starts.
+pub fn run_test_node(
+	task_executor: TaskExecutor,
+	key: Sr25519Keyring,
+	storage_update_func: impl Fn(),
+	boot_nodes: Vec<MultiaddrWithPeerId>,
+) -> Result<
+	PolkadotTestNode<
+		TaskManager,
+		impl PolkadotClient<Block, TFullBackend<Block>, polkadot_test_runtime::RuntimeApi>,
+	>,
+	ServiceError,
+> {
+	let config = node_config(storage_update_func, task_executor, key, boot_nodes)?;
+	let multiaddr = config.network.listen_addresses[0].clone();
+	let authority_discovery_enabled = false;
+	let (task_manager, client, handles, network, rpc_handlers) =
+		polkadot_test_new_full(config, None, None, authority_discovery_enabled, 6000)?;
+
+	let peer_id = network.local_peer_id().clone();
+	let multiaddr_with_peer_id = MultiaddrWithPeerId { multiaddr, peer_id };
+
+	let node = PolkadotTestNode {
+		task_manager,
+		client,
+		handles,
+		multiaddr_with_peer_id,
+		rpc_handlers,
+	};
+
+	Ok(node)
+}
+
+/// A Polkadot test node instance used for testing.
+pub struct PolkadotTestNode<S, C> {
+	/// TaskManager's instance.
+	pub task_manager: S,
+	/// Client's instance.
+	pub client: Arc<C>,
+	/// Node's handles.
+	pub handles: FullNodeHandles,
+	/// The `MultiaddrWithPeerId` to this node. This is useful if you want to pass it as "boot node" to other nodes.
+	pub multiaddr_with_peer_id: MultiaddrWithPeerId,
```

Comment:
```suggestion
	pub addr: MultiaddrWithPeerId,
```

You don't need to write out the name in lower case here